homeassistant/components/backup/manager.py

```diff
@@ -940,9 +917,63 @@ def _mkdir_and_generate_backup_contents(
                 )
         return (tar_file_path, tar_file_path.stat().st_size)
 
+    async def async_receive_backup(
+        self,
+        *,
+        agent_ids: list[str],
+        stream: AsyncIterator[bytes],
+        suggested_filename: str,
+    ) -> WrittenBackup:
+        """Receive a backup."""
+        temp_file = Path(self.temp_backup_dir, suggested_filename)
+
+        async_add_executor_job = self._hass.async_add_executor_job
+        await async_add_executor_job(make_backup_dir, self.temp_backup_dir)
+        f = await async_add_executor_job(temp_file.open, "wb")
+        try:
+            async for chunk in stream:
+                await async_add_executor_job(f.write, chunk)
+        finally:
+            await async_add_executor_job(f.close)
+
+        try:
+            backup = await async_add_executor_job(read_backup, temp_file)
```

Comment:
Why do we need to write the stream to disk, then read the file, then create a new stream that we pass to the agents? Can't we just keep the data as a stream, maybe decorating the stream if it needs to be reformatted, and pass it to the agents? Is it because we need to use the stream multiple times with different agents?

Reply:
It's to handle two cases:
- The backup we receive from the user's browser may not be valid, so we need to validate it
- The backup may be meant for more than one agent

Reply:
Right, but unless there's a size limitation that prohibits it, it feels like this could all be done in memory, without disk access at this time.

Reply:
I guess that's the problem, which makes it impossible to reuse the stream without having it all in memory.

Reply:
We have no idea how big the backup is, it could be many GB for all we know.